File: scala/src/main/scala/wordle/internal/HumanPlayer.scala
package wordle.internal

import wordle.internal.domain.{GuessFeedback, Player}

object HumanPlayer {
  def apply(): HumanPlayer = {
    new HumanPlayer
  }
}

class HumanPlayer extends Player {

  def guessWord(guessFeedback: Seq[GuessFeedback]): String = {
    print("Enter guess: ")
    Console.in.readLine().trim
  }
}

File: java/app/src/main/java/wordle/internal/HumanPlayer.java
package wordle.internal;

import java.util.List;
import java.util.Scanner;

public class HumanPlayer implements Player {

    private final Scanner scanner = new Scanner(System.in);

    @Override
    public String guessWord(List<GuessFeedback> guessFeedback) {
        return scanner.nextLine().trim();
    }
}
